content/_code-samples/secure-signing/py/sign-payment.py

```diff
@@ -0,0 +1,34 @@
+from xrpl.core import keypairs 
+
+# Define network client
+from xrpl.clients import JsonRpcClient
+JSON_RPC_URL = "https://s.altnet.rippletest.net:51234/"
+client = JsonRpcClient(JSON_RPC_URL)
+
+# Define signer address
+from xrpl.wallet import Wallet
+test_wallet_1 = Wallet(seed="shrPSF6vV3v3yoND9J3NeKks6M3xd")
+print(test_wallet_1.classic_address) # "raaFKKmgf6CRZttTVABeTcsqzRQ51bNR6Q"
+
+# Prepate the tranaaction
+from xrpl.ledger import get_latest_validated_ledger_sequence
+current_validated_ledger = get_latest_validated_ledger_sequence(client)
+min_ledger = current_validated_ledger + 1
+max_ledger = min_ledger + 20
```

Comment:
For offline signing you can skip this entire section (you have to either manually specify the `last_ledger_sequence`, as in the ripple-lib example, or omit it)